### PR TITLE
Contribution Guide

### PR DESCRIPTION
**Pull request by ashirrwad**
Added the contributors guide for easy understanding of the codebase. 
Fixed minor dev leftovers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,6 +133,99 @@ This should open the Fortran playground in your default web browser.
...
+  The Styling for the app is stored under `frontend/src/App.css`. All styles are applied from a single CSS file.
+  
+### Backend
+We use Flask to handle our backend operations, Docker-py is used to communicate with the docker engine on the server. The Dockerfile for the sandbox environment is stored under `backend/Docker`, it also contains the fpm.toml file for pre-building libraries on the container. 
+Functions:
+* edit_file(code, input, libs)
+  1. It's responsible for editing the files with user code and user input.
+  2. Editing fpm.toml file with library path is also done by this function. When adding a new library update this function with the library and it's path on the container. For eg:
+  `if "library-name" in libs:
+      fpm["dependencies"] = {'library-name' : {'path' : "libraries/library-path"}}
```
Comment (awvwgk) [inline README.md:211]: Use ```` ``` ```` backticks for multiline code blocks:

````suggestion
  ```
  if "library-name" in libs:
      fpm["dependencies"] = {'library-name' : {'path' : "libraries/library-path"}}
  ```
````

**Comment by ashirrwad** [comment]
Thanks for pointing out, fixed.

**Review by everythingfunctional — APPROVED** [review]
LGTM.

**Review by milancurcic — APPROVED** [review]
Thanks, @ashirrwad. I resolved the conflicts and put this guide in its own file (DESIGN.md) since the README was getting quite big and somewhat different purposes/audiences.

I noticed that it currently covers only the frontend component of the system. Do you plan to make a similar doc for the backend? I think it would be helpful. I'll go ahead and merge this, and we can tackle the rest in separate PRs.

**Pull request merged**